homeassistant/components/recorder/__init__.py

```diff
@@ -121,6 +124,25 @@ def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     instance.async_initialize()
     instance.start()
 
+    @asyncio.coroutine
+    def async_handle_purge_service(service):
```

Comment:
Add a optional parameter to set purge days. So it should be possible to overwrite the default purge day from config.